lib/board.rb

```diff
@@ -107,22 +107,21 @@ def render(show_ship=false)
     rows = Math.sqrt(@cells.keys.length).to_i
     columns = rows
 
+    rr = " "
+    #rr = rendered_rows
+
       (1..columns).to_a.each do |header|
-        print " #{header}"
+        rr += " #{header}"
       end
-      puts ' '
-
-      index = 0
+      rr += " \n"
 
-      ('A'..'D').to_a.each do |sider|
-        print "#{sider}"
-          @cells.keys[0..columns - 1].each do |key|
-          print " #{@cells[key].render(show_ship)}"
+      ('A'..'D').to_a.each_with_index do |sider, index|
+        rr += "#{sider}"
```

Comment:
Who knew string concatenation could be so useful for storing results of nested enumerators in string form!